app/lib/api/auth.ts

```diff
@@ -26,12 +30,20 @@ interface AuthorizationResponse {
     displayName: string;
     accountId: string;
 }
-export const authorizeWithAccessCode = async (accessCode: string): Promise<AuthorizationResponse> => {
+export const authorizeWithAccessCode = async (
+    accessCode: string, clientCode?: string,
+): Promise<AuthorizationResponse> => {
     const params = {
         code: accessCode,
-        redirect_uri: getRedirectUri(),
+        redirect_uri: getRedirectUri(!!clientCode),
+        clientCode,
     };
 
+    if (!clientCode) {
+        // make sure clientCode is only sent if it exists
+        delete params.clientCode;
+    }
```

Comment:
Might just be personal preference, but could be up for debate to not set the `clientCode` by default, then only add it to the params when it exists. Looks a bit cleaner than using `delete` imo:
```
    const params = {
        code: accessCode,
        redirect_uri: getRedirectUri(!!clientCode),
    };

    if (!!clientCode) {
        params.clientCode = clientCode;
    }
```